mmcv/fileio/file_client.py

```diff
@@ -410,6 +411,375 @@ def _list_dir_or_file(dir_path, list_dir, list_file, suffix,
                                  recursive)
 
 
+class AWSBackend(BaseStorageBackend):
+    """AWSBackend is Amazon Simple Storage Service(s3).
+
+    AWSBackend supports reading and writing data to aws s3.
+    It relies on awscli and boto3, you must install and run ``aws configure``
+    in advance to use it.
+
+    Args:
+        path_mapping (dict, optional): Path mapping dict from local path to
+            Petrel path. When ``path_mapping={'src': 'dst'}``, ``src`` in
+            ``filepath`` will be replaced by ``dst``. Default: None.
+
+    Examples:
+        >>> filepath = 's3://bucket/obj'
+        >>> client = AWSBackend()
+        >>> client.get(filepath1)  # get data from aws s3
+        >>> client.put(obj, filepath)
+    """
+
+    def __init__(self, path_mapping: Optional[dict] = None):
+        try:
+            import boto3
+            from botocore.exceptions import ClientError
+        except ImportError:
+            raise ImportError('Please install boto3 to enable AWSBackend.')
+
+        self._client = boto3.client('s3')
+        assert isinstance(path_mapping, dict) or path_mapping is None
+        self.path_mapping = path_mapping
+        # Use to parse bucket and obj_name
+        self.parse_bucket = re.compile('s3://(.+?)/(.+)')
+        self.check_exception = ClientError
+
+    def _map_path(self, filepath: Union[str, Path]) -> str:
```

Comment:
It seems that many functions such as `_map_path`, `_format_path` are implemented in subclasses. Could we implement them in base class?

Reply:
Yes, i will fix it.